paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -260,125 +259,137 @@ def paasta_mark_for_deployment(args):
+                             "running"
+                             .format(service, status.instance, cluster,
+                                     status.marathon.app_count))
                 continue
             if not git_sha.startswith(status.git_sha):
-                paasta_print("  {}.{} on {} doesn't have the right sha yet: {}".format(
-                    service, status.instance, cluster, status.git_sha))
-                results.append(False)
+                paasta_print("  {}.{} on {} doesn't have the right sha yet: {}"
+                             .format(service, instance, cluster,
+                                     status.git_sha))
                 continue
             if status.marathon.deploy_status != 'Running':
-                paasta_print("  {}.{} on {} isn't running yet: {}".format(
-                    service, status.instance, cluster, status.marathon.deploy_status))
-                results.append(False)
+                paasta_print("  {}.{} on {} isn't running yet: {}"
+                             .format(service, instance, cluster,
+                                     status.marathon.deploy_status))
                 continue
-            if status.marathon.expected_instance_count != status.marathon.running_instance_count:
-                paasta_print("  {}.{} on {} isn't scaled up yet, has {} out of {}".format(
-                    service, status.instance, cluster, status.marathon.running_instance_count,
-                    status.marathon.expected_instance_count))
-                results.append(False)
+            if (status.marathon.expected_instance_count !=
+                    status.marathon.running_instance_count):
+                paasta_print("  {}.{} on {} isn't scaled up yet, "
+                             "has {} out of {}"
+                             .format(service, instance, cluster,
+                                     status.marathon.running_instance_count,
+                                     status.marathon.expected_instance_count))
                 continue
-            paasta_print("Complete: {}.{} on {} looks 100% deployed at {} instances on {}".format(
-                service, status.instance, cluster, status.marathon.running_instance_count, status.git_sha))
-            results.append(True)
+            paasta_print("Complete: {}.{} on {} looks 100% deployed at {} "
+                         "instances on {}"
+                         .format(service, instance, cluster,
+                                 status.marathon.running_instance_count,
+                                 status.git_sha))
+            result.append(instance)
 
-    return sum(results)
+    return result
 
 
 def wait_for_deployment(service, deploy_group, git_sha, soa_dir, timeout):
-    cluster_map = get_cluster_instance_map_for_service(soa_dir, service, deploy_group)
+    cluster_map = get_cluster_instance_map_for_service(soa_dir, service,
```

Comment:
please kwargs this. I'm surprised to see soa_dir as the first arg, not that it *matters* but I just hate arg order bugs.